app/questionnaire/questionnaire_schema.py

```diff
@@ -1,80 +1,61 @@
     def __init__(self, questionnaire_json):
         self.json = questionnaire_json
-        self._groups_by_id = self._get_groups_by_id()
-        self._blocks_by_id = self._get_blocks_by_id()
-        self._answers_by_id = self._get_answers_by_id()
-        self.error_messages = self._get_error_messages()
-        self.aliases = self._get_aliases()
+        self._parse_schema()
 
-    def _get_groups_by_id(self):
-        groups = {}
-        for group in self.json.get('groups', []):
-            groups[group.get('id')] = group
+    @property
+    def sections(self):
+        return self._sections_by_id.values()
 
-        return groups
+    @property
+    def groups(self):
+        return self._groups_by_id.values()
 
-    def _get_blocks_by_id(self):
-        blocks = {}
-        for group in self.json.get('groups', []):
-            for block in group['blocks']:
-                blocks[block.get('id')] = block
+    @property
+    def blocks(self):
+        return self._blocks_by_id.values()
 
-        return blocks
+    @property
+    def questions(self):
+        return self._questions_by_id.values()
 
-    def _get_answers_by_id(self):
-        answers = {}
-        for block in self.get_blocks():
-            for question in self.get_questions_for_block(block):
-                for answer in question['answers']:
-                    answers[answer['id']] = answer
+    @property
+    def answers(self):
+        return self._answers_by_id.values()
 
-        return answers
+    def get_section(self, section_id):
+        return self._sections_by_id.get(section_id)
 
-    def _get_error_messages(self):
-        messages = error_messages.copy()
-        if 'messages' in self.json:
-            for key, message in self.json['messages'].items():
-                messages[key] = message
+    def get_group(self, group_id):
+        return self._groups_by_id.get(group_id)
 
-        return messages
+    def get_block(self, block_id):
+        return self._blocks_by_id.get(block_id)
 
-    def _get_aliases(self):
-        aliases = {}
-        for block in self.get_blocks():
-            for question in self.get_questions_for_block(block):
-                for answer in question['answers']:
-                    if 'alias' in answer:
-                        if answer['alias'] in aliases:
-                            raise Exception('Duplicate alias found: ' + answer['alias'])
-                        aliases[answer['alias']] = {
-                            'answer_id': answer['id'],
-                            'repeats': answer['type'] == 'Checkbox' or question['type'] == 'RepeatingAnswer',
-                        }
+    def get_question(self, question_id):
+        return self._questions_by_id.get(question_id)
 
-        return aliases
+    def get_answer(self, answer_id):
+        return self._answers_by_id.get(answer_id)
 
-    def get_sections(self):
+    def get_sections_for_navigation(self):
```

Comment:
i think we should steer away from making this class supply data for a specific use.
The navigation class calling it should do the work.
or is there any? isn't this just returning a list of sections?

Reply:
Good point, just discussed this separately and there's no value in definition sections explicitly for navigation.